contrib/msggen/msggen/model.py

```diff
@@ -204,15 +204,14 @@ def from_js(cls, js, path):
             props = var.get('properties', None)
             if isinstance(props, dict):
                 for k, v in props.items():
-                    if k not in properties:
+                    if k not in properties or k in properties and properties[k] == {}:
                         properties[k] = v
             var = a.get('else', {})
             props = var.get('properties', None)
             if isinstance(props, dict):
                 for k, v in props.items():
-                    if k not in properties:
+                    if k not in properties or k in properties and properties[k] == {}:
```

Comment:
This needs parentheses, otherwise the `and` and `or` precedence is confusing. The first part of the statement is even a tautology (a or not a), so I'm not sure this matches what you are expecting here.